src/main/java/uk/gov/hmcts/reform/sscs/service/DwpAddressLookupService.java

```diff
@@ -92,113 +91,88 @@ private Address buildAddress(OfficeAddress dwpAddress) {
-            if (officeMapping.map(m -> m.getMapping().getDwpRegionCentre() != null).orElse(false)) {
-                return officeMapping.map(mapping -> mapping.getMapping().getDwpRegionCentre()).orElse(null);
-            } else {
-                return officeMapping.map(mapping -> mapping.getMapping().getCcd()).orElse(null);
-            }
-        } catch (BenefitMappingException e) {
-            return null;
-        }
-    }
-
     public Optional<OfficeMapping> getDwpMappingByOffice(String benefitType, String dwpIssuingOffice) {
         log.info("looking up officeAddress for benefitType {} and dwpIssuingOffice {}", benefitType, dwpIssuingOffice);
 
         if (equalsIgnoreCase(dwpIssuingOffice, TEST_HMCTS_ADDRESS)) {
             return Optional.of(dwpMappings.getTestHmctsAddress());
         }
-        return findBenefitByShortName(benefitType)
-                .flatMap(b -> b.getOfficeMappings().apply(this, dwpIssuingOffice));
-    }
-
-    public Optional<OfficeMapping> carersAllowanceOfficeMapping(@SuppressWarnings("unused") String dwpIssuingOffice) {
-        return Optional.of(dwpMappings.getCarersAllowance());
-    }
-
-    public Optional<OfficeMapping> dlaOfficeMapping(String dwpIssuingOffice) {
-        return getOfficeMappingByDwpIssuingOffice(dwpIssuingOffice, dwpMappings.getDla());
-    }
-
-    public Optional<OfficeMapping> attendanceAllowanceOfficeMapping(String dwpIssuingOffice) {
-        return getOfficeMappingByDwpIssuingOffice(dwpIssuingOffice, dwpMappings.getAttendanceAllowance());
-    }
-
-    public Optional<OfficeMapping> bereavementBenefitOfficeMapping(@SuppressWarnings("unused") String dwpIssuingOffice) {
-        return Optional.of(dwpMappings.getBereavementBenefit());
-    }
-
-    public Optional<OfficeMapping> iidbOfficeMapping(String dwpIssuingOffice) {
-        return getOfficeMappingByDwpIssuingOffice(dwpIssuingOffice, dwpMappings.getIidb());
-    }
-
-    public Optional<OfficeMapping> ucOfficeMapping(@SuppressWarnings("unused") String dwpIssuingOffice) {
-        return Optional.of(dwpMappings.getUc());
-    }
-
-    public Optional<OfficeMapping> esaOfficeMapping(String dwpIssuingOffice) {
-        return getOfficeMappingByDwpIssuingOffice(dwpIssuingOffice, dwpMappings.getEsa());
-    }
 
-    public Optional<OfficeMapping> pipOfficeMapping(String dwpIssuingOffice) {
-        String dwpIssuingOfficeStripped = ofNullable(substringBetween(dwpIssuingOffice,"(", ")"))
-                .orElse(dwpIssuingOffice.replaceAll("\\D+",""));
-
-        if (isEmpty(dwpIssuingOfficeStripped)) {
-            dwpIssuingOfficeStripped = dwpIssuingOffice;
+        //PIP search code
+        String dwpIssuingOfficeStripped = null;
+        if (findBenefitByShortName(benefitType).map(b -> b == PIP).orElse(false)) {
+            dwpIssuingOfficeStripped = ofNullable(substringBetween(dwpIssuingOffice, "(", ")"))
+                    .orElse(dwpIssuingOffice == null ? "" : dwpIssuingOffice.replaceAll("\\D+", ""));
```

Comment:
It takes a while to work out what 
```java
if (findBenefitByShortName(benefitType).map(b -&gt; b == PIP).orElse(false)) {            dwpIssuingOfficeStripped = ofNullable(substringBetween(dwpIssuingOffice, "(", ")"))                    .orElse(dwpIssuingOffice == null ? "" : dwpIssuingOffice.replaceAll("\\D+", ""));
```
does. A descriptive method would make it clearer

Reply:
Refactored